src/binding-hash.h

```diff
@@ -18,34 +18,47 @@
 #define WABT_BINDING_HASH_H_
 
 #include "common.h"
-#include "vector.h"
+
+#include <string>
+#include <vector>
+#include <unordered_map>
 
 namespace wabt {
 
 struct Binding {
+  explicit Binding(int index) : index(index) {
+    WABT_ZERO_MEMORY(loc);
+  }
+  Binding(const Location& loc, int index) : loc(loc), index(index) {}
+
   Location loc;
-  StringSlice name;
   int index;
 };
 
-struct BindingHashEntry {
-  Binding binding;
-  struct BindingHashEntry* next;
-  struct BindingHashEntry* prev; /* only valid when this entry is unused */
-};
-WABT_DEFINE_VECTOR(binding_hash_entry, BindingHashEntry);
+class BindingHash : public std::unordered_multimap<std::string, Binding> {
```

Comment:
I'm a bit hesitant about using C++ containers as a base class, because it doesn't have a virtual destructor. However, this is only important if you are going to cast pointers to a BindingHash to its base class, and then delete it.

Considering how you are using this class, it should be acceptable. However, I might add a comment that one shouldn't cast pointers of this class to the C++ container base.